spec/enumerables_spec.rb

```diff
@@ -0,0 +1,242 @@
+require './app'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3] }
+  let(:block) { proc { |x| x * 2 } }
+  let(:inject_block) { proc { |prod, n| prod * n } }
+  let(:block_with_index) { proc { |num, i| puts "Num: #{num}, index: #{i}" } }
+  let(:range) { (1..5) }
+  let(:my_hash) { { first_name: 'somoye', second_name: 'nusret' } }
+  let(:hash_block) { proc { |key, value| puts "#{key} is #{value}" } }
+  let(:rg) { '/d/' }
+
+  describe '#my_each' do
+    it 'it iterates the array and return Enumerator if no block is given' do
+      expect(array.my_each).to be_a(Enumerator)
+    end
+
+    it 'Iterates through the array and executes block' do
+      expect(array.my_each(&block)).to eq(array.each(&block))
+    end
+
+    it 'Iterates through the range and executes block' do
+      expect(range.my_each(&block)).to eq(range.each(&block))
+    end
+
+    it 'Iterates through the hash and executes block' do
+      expect(my_hash.my_each(&block)).to eq(my_hash.each(&block))
+    end
+
+    it 'Iterates through the hash and executes block' do
+      expect(my_hash.my_each(&hash_block)).to eq(my_hash.each(&hash_block))
+    end
+  end
```

Comment:
You can create a context for all these tests instead of repeating `executes block` and just add the `it` statement for each type.
Please apply the same to all tests where you repeat this.